Bluewire.RepositoryLinter/RepositoryExplorer.cs

```diff
@@ -97,10 +107,39 @@ private async Task<ProjectFile> ReadProjectFile(string path, Stream stream)
             }
         }
 
-        private PackageReference ReadPackageReference(XElement element)
+        private IEnumerable<PackageReference> ReadPackageReferences(IEnumerable<XElement> packageReferences)
+        {
+            foreach (var element in packageReferences)
+            {
+                var name = element.Attribute("Include")?.Value;
+                if (name == null)
+                {
+                    if (element.Attribute("Update") != null) continue;
+                    throw new ArgumentException("PackageReference element has no Include or Update attribute.");
+                }
+                var versionElement = element.Element("Version");
+                if (versionElement?.Value != null)
+                {
+                    yield return new PackageReference(name, versionElement.Value);
+                    continue;
+                }
+                var versionAttribute = element.Attribute("Version");
+                if (versionAttribute?.Value != null)
+                {
+                    yield return new PackageReference(name, versionAttribute.Value);
+                    continue;
+                }
+                throw new ArgumentException("PackageReference element has no Version element or attribute.");
+            }
+        }
+
+        private PackageReference? ReadPackageReference(XElement element)
         {
             var name = element.Attribute("Include")?.Value;
-            if (name == null) throw new ArgumentException("PackageReference element has no Include attribute.");
+            if (name == null)
+            {
+                throw new ArgumentException("PackageReference element has no Include or Update attribute.");

```

Comment:
Exception message states  `Update` when only `Include` is checked?

Reply:
I think that's dead code actually. Must've forgotten to delete that copy.